src/core/api.c

```diff
@@ -616,6 +621,204 @@ mender_api_publish_deployment_status(const char *id, mender_deployment_status_t
         ret = MENDER_FAIL;
     }
 
+END:
+
+    /* Release memory */
+    mender_free(response);
+    mender_free(path);
+    mender_free(payload);
+    cJSON_Delete(json_payload);
+
+#ifdef CONFIG_MENDER_DEPLOYMENT_LOGS
+    /* Do this after we have released memory above, potentially giving us some
+       extra room we may need. */
+    if ((MENDER_OK == ret) && (MENDER_DEPLOYMENT_STATUS_FAILURE == deployment_status)) {
+        /* Successfully reported a deployment failure, upload deployment
+           logs.  */
+        if (MENDER_OK != mender_api_publish_deployment_logs(id)) {
+            mender_log_error("Failed to publish deployment logs");
+        }
+    }
+#endif /* CONFIG_MENDER_DEPLOYMENT_LOGS */
+
+    return ret;
+}
+
+#ifdef CONFIG_MENDER_DEPLOYMENT_LOGS
+static void
+append_depl_log_msg(char *msg, void *ctx) {
```

Comment:
This function we should unit test

Reply:
I agree. But have we agreed on how we'll unit test `static` functions?

Reply:
We don't have tests for static functions directly (not yet, at least). But since we have now private and public header files, that is a easy way to expose the bits we want to test.

Reply:
> that is a easy way to expose the bits we want to test

It's not, unfortunately. It's not only about headers, you'd still have to drop `static` from the function declaration/definition and it would get into the linker namespace, potentially causing conflicts and unexpected behavior.

Reply:
I meant both: removing the static and moving it into the private header (with a prefix). The point is that it is still out of the public header file so it should not be on the way for users or other compile units.

Reply:
Yeah, but then we need to prefix the name with `mender_`. And I generally don't like the idea of making functions non-static just because we want to unit-test them.